src/generic_factory.h

```diff
@@ -844,6 +1011,51 @@ class generic_typed_reader
             }
         }
 
+        // These two functions are effectively handle_relative
+        // but they need to use the reader, so they must be here
```

Comment:
Why is this true for relative but not proportional?

Reply:
Proportional is only reading a float, relative is reading the same type of value.

Reply:
And now there's a comment saying so.